guidelines/dxp/extracting_js_from_jsps.md

```diff
@@ -0,0 +1,41 @@
+# Extracting JS from JSPs
+
+As we're modernizing our codebase across DXP we are starting to encounter cases where we are touching JS code inside JSPs and often times we have to remind ourselves how best to tackle that.
+
+We have two options when extracting JS from JSPs:
+
+-   `<liferay-frontend:component>` - good old JS that doesn't render UI
+-   `<react:component>`- markup with interactive behaviour
+
+The `context` and `props` arguments passed to the tag are of type `Map<String, Object>` built using the `HashMapBuilder.<String, Object>`.
+
+## Using `<liferay-frontend:component>`
+
+Whenever you have some simpler JS code inside the `<aui:script>` you most likely want to use this tag. It allows you to easily send backend data (`context`) into the scope of your JS function, it separates concerns and leads to cleaner code.
+
+### Code Example
+
+```javascript
+<liferay-frontend:component
+	componentId="Foo" // If your JS code is in form of a Class, this is where you specify the Class name
+	context="<%= context %>" // Data you want to send from the JSP to the JS
+	module="foo-bar/js/foo.es" // Relative path to your new JS file
```

Comment:
nit: we are moving away from `.es` as they aren't necessary anymore and so it'd probably be best to have `.js` here